### PR TITLE
update controller runtime to v0.15

### PR DESCRIPTION
**Pull request by ranakan19**
Updates k8s dependencies to 0.28 and controller-runtime to v0.15.

All the other PRs update k8s dependencies to 0.27 but this one updates to 0.28 because of the following error:

`../../../../pkg/mod/k8s.io/kubectl@v0.25.0/pkg/describe/describe.go:2667:65: cannot use ing.Status.LoadBalancer (variable of type "k8s.io/api/networking/v1".IngressLoadBalancerStatus) as "k8s.io/api/core/v1".LoadBalancerStatus value in argument to loadBalancerStatusStringer`

and kubectl 0.28 was the only comptabile version. 

With the update to controller-runtime v0.15 (changes in release detailed [here](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0)), this PR has following changes to combat the breaking changes:
- NewClient creation does not fail, so tests where the client is expected to fail specific api call needs to be made
- replace apimachinery/runtime client with controller-runtime/client
 
Changes wrt - https://issues.redhat.com/browse/SANDBOX-675
DO NOT MERGE (All the PRs related to version updates need to be merged at once, but these PR are for early feedback so that there are not a lot of changes to review at once)

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -42,10 +42,19 @@ func TestNewClientOK(t *testing.T) {
...
+	require.NoError(t, err)
+	assert.NotNil(t, cl)
 	// then
+	testObj := &toolchainv1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "john-doe",
+			Namespace: "default",
+		},
+	}
+	_, err = cl.IsObjectNamespaced(testObj)
```
Comment (alexeykazakov) [inline pkg/client/client_test.go:54]: Let's add a comment explaining why we check if the object is namespaced (we just need to make ANY request to the actual API to trigger the error for an incorrectly configured client).

**Review by alexeykazakov — APPROVED** [review]
Looks good. Just one comment to address.

File: pkg/client/client_test.go [hunk 2]
```diff
@@ -42,10 +42,23 @@ func TestNewClientOK(t *testing.T) {
...
+	}
+	// no specific reason to check if object is namespaced, ANY request to the actual api would trigger error indicating incorrect configuration of client
+	_, err = cl.IsObjectNamespaced(testObj)
 	require.Error(t, err)
-	assert.Nil(t, cl)
+	// actual error is "failed to get restmapping: failed to get server groups: Get \"https://fail-cluster.com/api?timeout=1m0s\": dial tcp: lookup fail-cluster.com: no such host"
+	require.ErrorContains(t, err, "dial tcp: lookup fail-cluster.com")
+	// for ci the error message is dial tcp: lookup fail-cluster.com on 127.0.0.53:53: no such host, could go the regex way or string submatching with wildcards
+	// having two separate checks for different substrings is an easy fix
+	require.ErrorContains(t, err, "no such host")
```
Comment (ranakan19) [inline pkg/client/client_test.go:61]: I did consider only checking for "no such host", but I do see value in checking for the host name as well - which we are setting explicitly in the test above.

File: pkg/client/client_test.go (hunk 1 above)
Comment (ranakan19) [reply pkg/client/client_test.go:55]: done in [6bb562d](https://github.com/kubesaw/ksctl/pull/86/commits/6bb562dc8b23fb3b5a793191204662393a857418)

**Pull request merged**